### PR TITLE
HW test failure slacking

### PR DESCRIPTION
**Pull request by mnokka-unikie**
Reintroduced hw test errors slacking and merging to latest main as a new fresh branch

### CODE REVIEW

File: utils.groovy [hunk 1]
```diff
@@ -233,7 +233,7 @@ def ghaf_hw_test(String flakeref, String device_config, String testset='_boot_')
     // Add a link to failed test job(s) on the calling pipeline
     def test_href = "<a href=\"${job.absoluteUrl}\">⛔ ${flakeref_trimmed}</a>"
     currentBuild.description = "${currentBuild.description}<br>${test_href}"
-    return
+    return flakeref
```
Comment (cazfi) [inline utils.groovy:236]: This says it's a flake reference...

File: utils.groovy [hunk 2]
```diff
@@ -421,7 +421,15 @@ def create_parallel_stages(List<Map> targets, String testset='_boot_bat_perf_',
       if (testset != null && it.hwtest_device != null) {
         stage("Test ${displayName}") {
           script {
-            ghaf_hw_test(targetAttr, it.hwtest_device, testset)
+            errorstatus=ghaf_hw_test(targetAttr, it.hwtest_device, '_boot_bat_perf_')
```
Comment (cazfi) [inline utils.groovy:424]: ...this says it's an error status...

File: utils.groovy [hunk 3]
```diff
@@ -421,7 +421,15 @@ def create_parallel_stages(List<Map> targets, String testset='_boot_bat_perf_',
...
+            println ("Test creation and execution done")
+            if (errorstatus==null) {
+              println("Test was OK")
+            }
+            else {
+              failedHWTests.add(errorstatus)
+              println ("ERROR FOUND, target: ${errorstatus}")
```
Comment (cazfi) [inline utils.groovy:431]: ...and these claim it to be a target.

Rework variable naming etc to make it clear what it is.

**Review by cazfi — CHANGES_REQUESTED** [review]

File: utils.groovy (hunk 3 above)
Comment (mnokka-unikie) [reply utils.groovy]: using same naming now, forced pushed change

**Comment by mnokka-unikie** [comment]
force pushed change, using same var naming now

File: utils.groovy [hunk 4]
```diff
@@ -421,7 +421,15 @@ def create_parallel_stages(List<Map> targets, String testset='_boot_bat_perf_',
       if (testset != null && it.hwtest_device != null) {
         stage("Test ${displayName}") {
           script {
-            ghaf_hw_test(targetAttr, it.hwtest_device, testset)
+            flakeref=ghaf_hw_test(targetAttr, it.hwtest_device, '_boot_bat_perf_')
+            println ("Test creation and execution done")
+            if (errorstatus==null) {
```
Comment (cazfi) [inline utils.groovy:426]: Should this be flakeref?

File: utils.groovy [hunk 5]
```diff
@@ -421,7 +421,15 @@ def create_parallel_stages(List<Map> targets, String testset='_boot_bat_perf_',
...
+              println("Test was OK")
+            }
+            else {
+              failedHWTests.add(errorstatus)
```
Comment (cazfi) [inline utils.groovy:430]: Should this be flakeref?

File: utils.groovy (hunk 4 above)
Comment (mnokka-unikie) [reply utils.groovy]: indeed, fixed directly

File: utils.groovy (hunk 5 above)
Comment (mnokka-unikie) [reply utils.groovy]: indeed, fixed directly

**Review by cazfi — APPROVED** [review]

**Pull request merged**